consensus/core/src/synchronizer.rs

```diff
@@ -975,21 +975,16 @@ impl<C: NetworkClient, V: BlockVerifier, D: CoreThreadDispatcher> Synchronizer<C
             .take(MAX_PEERS * context.parameters.max_blocks_per_fetch)
             .collect::<Vec<_>>();
 
-        #[allow(unused_mut)]
+        #[cfg_attr(test, expect(unused_mut))]
         let mut peers = context
             .committee
             .authorities()
             .filter_map(|(peer_index, _)| (peer_index != context.own_index).then_some(peer_index))
             .collect::<Vec<_>>();
 
-        // TODO: probably inject the RNG to allow unit testing - this is a work around
-        // for now.
-        cfg_if::cfg_if! {
-            if #[cfg(not(test))] {
```

Comment:
I don't understand what this was doing

Reply:
I mean, exactly the same I assume, but why this syntax

Reply:
me neither 😆